nilearn/mass_univariate/tests/test_permuted_least_squares.py

```diff
@@ -14,106 +16,159 @@
 from scipy import stats
 from sklearn.utils import check_random_state
 
+RANDOM_STATE = 0
+
+N_SAMPLES = 50
+
+N_COVARS = 2
+
+
+def mean_squared_error(df, h0_intercept):
+    mse = np.mean(
+        (
+            stats.t(df).cdf(np.sort(h0_intercept))
+            - np.linspace(0, 1, h0_intercept.size + 1)[1:]
+        )
+        ** 2
+    )
+    return mse
+
+
+def ks_stat_and_mse(df, h0_intercept, kstest_pvals_list, mse_list):
+    """Run Kolmogorov-Smirnov test and compute Mean Squared Error"""
+    kstest_pval = stats.kstest(h0_intercept, stats.t(df).cdf)[1]
+    kstest_pvals_list.append(kstest_pval)
+    mse = mean_squared_error(df=df, h0_intercept=h0_intercept)
+    mse_list.append(mse)
+
+
+def permuted_ols_no_intercept(tested_var, target_var, n_perm, i):
+    n_regressors = 1
+    _, _, h0 = permuted_ols(
```

Comment:
Most of the tests in this module use the legacy ouput for this function and this causes truck load of deprecation warnings in testing.

Should I update the ouputs in most function but those that check the actual warning?